packages/cache-url/lib/AmpCurlUrlBrowserSha256.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @param {string} str The string to convert to sha256
+ * @return {!Promise<string>}
+ * @private
+ */
+export default function ampCurlUrlBrowserSha256(str) {
+  // Transform the string into an arraybuffer.
+  const buffer = new TextEncoder('utf-8').encode(str);
```

Comment:
Not sure what the intended browser support matrix is, but `TextEncoder` isn't available for Edge or IE.

https://caniuse.com/#search=TextEncoder

Reply:
I copied this from the AMP Viewer code, as I just wanted to migrate it here. So It should have already been approved 😮 Though, if you feel like we should use something else let me know 😄